### PR TITLE
[consensus] only echo timeout when no timeout yet

### PR DESCRIPTION
**Pull request by zekun000**
### Test Plan

---
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2187)

### CODE REVIEW

File: consensus/src/liveness/round_state.rs [hunk 1]
```diff
@@ -200,6 +200,11 @@ impl RoundState {
         }
     }
 
+    /// Return if already vote for timeout
```
Comment (msmouse) [inline consensus/src/liveness/round_state.rs:203]: "voted"

**Comment by bchocho** [comment]
Is this a small optimization or were there issues with the previous approach?

**Comment by zekun000** [comment]
it's a bug from https://github.com/aptos-labs/aptos-core/pull/2095 it'll double broadcast if it already timeout

**Review by davidiw — APPROVED** [review]

**Comment by ikabiljo** [comment]
@zekun000 - can we have a unit test for this? If that is complicated to do - can you explain why?

**Comment by zekun000** [comment]
I was lazy during the weekend, I'll add one

**Pull request merged**